### PR TITLE
add-annotation-validation-for-awcluster

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/roadmap/issues/216 and https://github.com/giantswarm/roadmap/issues/217

validation for awscluster to ensure proper values

AWSCluster validation was misssing so adding the wiring as well

### CODE REVIEW

File: pkg/aws/awsmachinedeployment/validate_awsmachinedeployment.go [hunk 1]
```diff
@@ -125,12 +125,12 @@ func (v *Validator) MachineDeploymentLabelMatch(awsMachineDeployment infrastruct
 
 func (v *Validator) MachineDeploymentAnnotationMaxBatchSizeIsValid(awsMachineDeployment infrastructurev1alpha2.AWSMachineDeployment) error {
 	if maxBatchSize, ok := awsMachineDeployment.GetAnnotations()[aws.AnnotationUpdateMaxBatchSize]; ok {
-		if !aws.MaxBatchSizeIsValid(maxBatchSize, awsMachineDeployment.Spec.NodePool.Scaling.Max) {
-			v.logger.Log("level", "debug", "message", fmt.Sprintf("AWSMachineDeployment annotation '%s' value '%s' is not valid. Allowed value is either positive integer number smaller than number of nodes or decimal number between 0 and 1.0 defining percentage of nodes",
+		if !aws.MaxBatchSizeIsValid(maxBatchSize) {
```
Comment (calvix) [inline pkg/aws/awsmachinedeployment/validate_awsmachinedeployment.go:128]: I figured out that I does not make sense to limit the maximum number for maxBatchSize as if you set it on AWSCluster you have no idea how each nodePool is big and AWS accept values that can be over the max number of nodes, so removing this constriction from both AWSCluster and AWSMachine deployment

File: pkg/aws/common.go [hunk 2]
```diff
@@ -51,15 +51,15 @@ func IsValidMasterReplicas(replicas int) bool {
...
 // float value is used as ratio of a total worker count
-func MaxBatchSizeIsValid(value string, maxWorkers int) bool {
+func MaxBatchSizeIsValid(value string) bool {
 	// try parse an integer
 	integer, err := strconv.Atoi(value)
 	if err == nil {
-		// check if the value is bigger than zero but lower-or-equal to maximum number of workers
-		if integer > 0 && integer <= maxWorkers {
+		// check if the value is bigger than zero but lower-or-equal
+		if integer > 0 {
```
Comment (calvix) [inline pkg/aws/common.go:62]: I figured out that I does not make sense to limit the maximum number for maxBatchSize as if you set it on AWSCluster you have no idea how each nodePool is big and AWS accept values that can be over the max number of nodes, so removing this constriction from both AWSCluster and AWSMachine deployment

File: helm/aws-admission-controller/templates/webhook.yaml [hunk 3]
```diff
@@ -95,6 +95,25 @@ metadata:
...
+      caBundle: Cg==
+    rules:
+    - apiGroups: ["infrastructure.giantswarm.io"]
+      resources:
+        - awsclusters
+      apiVersions:
+        - v1alpha2
+      operations:
+        - CREATE
+        - UPDATE
```
Comment (calvix) [inline helm/aws-admission-controller/templates/webhook.yaml:116]: adding new webhook for awscluster validation

File: pkg/aws/awscluster/validate_awscluster.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	var err error
+
+	if _, _, err := validator.Deserializer.Decode(request.Object.Raw, nil, &awsCluster); err != nil {
+		return false, microerror.Maskf(parsingFailedError, "unable to parse awscluster: %v", err)
+	}
+
+	err = v.AWSClusterAnnotationMaxBatchSizeIsValid(awsCluster)
+	if err != nil {
+		return false, microerror.Mask(err)
+
```
Comment (anvddriesch) [inline pkg/aws/awscluster/validate_awscluster.go:49]:
```suggestion
```

File: pkg/aws/awscluster/validate_awscluster.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	}
+
+	err = v.AWSClusterAnnotationPauseTimeIsValid(awsCluster)
+	if err != nil {
+		return false, microerror.Mask(err)
+
```
Comment (anvddriesch) [inline pkg/aws/awscluster/validate_awscluster.go:55]:
```suggestion
```

**Review by anvddriesch — APPROVED** [review]

File: pkg/aws/common.go [hunk 6]
```diff
@@ -51,15 +51,15 @@ func IsValidMasterReplicas(replicas int) bool {
...
 // a float between 0 < x <= 1
 // float value is used as ratio of a total worker count
-func MaxBatchSizeIsValid(value string, maxWorkers int) bool {
+func MaxBatchSizeIsValid(value string) bool {
 	// try parse an integer
 	integer, err := strconv.Atoi(value)
 	if err == nil {
-		// check if the value is bigger than zero but lower-or-equal to maximum number of workers
-		if integer > 0 && integer <= maxWorkers {
+		// check if the value is bigger than zero but lower-or-equal
```
Comment (njuettner) [inline pkg/aws/common.go:61]:
```suggestion
		// check if the value is bigger than zero
```

**Review by njuettner — APPROVED** [review]

**Pull request merged**